src/DynamoCore/Graph/Nodes/CodeBlockUtils.cs

```diff
@@ -265,6 +266,72 @@ internal static IOrderedEnumerable<KeyValuePair<string, int>> GetDefinitionLineI
 
             return locationMap.OrderBy(p => p.Value);
         }
+
+        /// <summary>
+        /// Extracts cleaned individual code expressions as tooltip strings from a full code block.
+        /// </summary>
+        internal static List<string> GetCleanedCodeExpressionsForTooltips(string code)
+        {
+            if (string.IsNullOrWhiteSpace(code))
+                return new List<string>();
+
+            // Filter out full-line comments and remove inline comments
+            var cleanedCode = string.Join(" ",
+                code.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries)
+                    .Where(line => !line.TrimStart().StartsWith("//"))
+                    .Select(line =>
+                    {
+                        var idx = line.IndexOf("//");
+                        return idx >= 0 ? line.Substring(0, idx) : line;
+                    }));
+
+            // Split by semicolon and return trimmed expressions
+            return cleanedCode
+                .Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries)
+                .Select(expr => expr.Trim())
+                .Where(expr => !string.IsNullOrEmpty(expr))
+                .Select(expr => expr + ";")
+                .ToList();
+        }
```

Comment:
I'm afraid this function needs to be entirely rewritten, as it shouldn't be using custom string manipulation when we have a full-fledged parser for DesignScript code and syntax. Will send another PR to clean this up.